src/test/java/org/folio/rest/impl/EholdingsProvidersImplTest.java

```diff
@@ -0,0 +1,129 @@
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.folio.config.cache.RMAPIConfigurationCache;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.folio.rest.util.RestConstants;
+import org.folio.util.TestUtil;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+@RunWith(VertxUnitRunner.class)
+public class EholdingsProvidersImplTest {
+  private static final Header TENANT_HEADER = new Header(RestConstants.OKAPI_TENANT_HEADER, "fs");
+  private static final Header TOKEN_HEADER = new Header(RestConstants.OKAPI_TOKEN_HEADER, "TEST_OKAPI_TOKEN");
+  private static final String STUB_CUSTOMER_ID = "TEST_CUSTOMER_ID";
+  private static RequestSpecification spec;
+  private static int port;
+  private static String host;
+
+  @org.junit.Rule
+  public WireMockRule userMockServer = new WireMockRule(
+    WireMockConfiguration.wireMockConfig()
+      .dynamicPort()
+      .notifier(new ConsoleNotifier(true)));
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) {
+    Vertx vertx = Vertx.vertx();
+    vertx.exceptionHandler(context.exceptionHandler());
+    port = NetworkUtils.nextFreePort();
+    host = "http://localhost";
+
+    DeploymentOptions restVerticleDeploymentOptions = new DeploymentOptions()
+      .setConfig(new JsonObject().put("http.port", port));
+    vertx.deployVerticle(RestVerticle.class.getName(), restVerticleDeploymentOptions, context.asyncAssertSuccess());
+
+    spec = new RequestSpecBuilder()
+      .setBaseUri(host + ":" + port)
+      .build();
+  }
+
+  @Before
+  public void setUp() {
+    RMAPIConfigurationCache.getInstance().invalidate();
+  }
+
+  @Test
+  public void shouldReturnProvidersOnGet() throws IOException, URISyntaxException {
+    String wiremockUrl = host + ":" + userMockServer.port();
+    TestUtil.mockConfiguration("responses/configuration/get-configuration.json", wiremockUrl);
+    WireMock.stubFor(
+      WireMock.get(new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/vendors.*"), true))
+        .willReturn(new ResponseDefinitionBuilder()
+          .withBody(TestUtil.readFile("responses/rmapi/vendors/get-vendors-response.json"))));
+
+    RestAssured.given()
+      .spec(spec).port(port)
+      .header(new Header(RestConstants.OKAPI_URL_HEADER, wiremockUrl))
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .when()
+      .get("eholdings/providers?q=e&page=1&sort=name")
+      .then()
+      .statusCode(200)
+      .body("meta.totalResults" , notNullValue())
+      .body("data[0].type", equalTo("providers"))
+      .body("data[0].id", notNullValue())
+      .body("data[0].attributes.name", notNullValue());
+  }
```

Comment:
Just wondering if we should check actual sample results here instead of checking for noNullValue?